src/main.lua

```diff
@@ -355,39 +355,9 @@ end, function(msg)
 
 	local msgId = msg.Id
 	print("Pruning state at timestamp: " .. msgTimestamp)
-	-- we need to be concious about deep copying here, as it could consume a large amount of memory. so long as we are pruning effectively, this should be fine
-	local previousState = {
-		Vaults = utils.deepCopy(Vaults),
-		GatewayRegistry = utils.deepCopy(GatewayRegistry),
-		NameRegistry = utils.deepCopy(NameRegistry),
-		Epochs = utils.deepCopy(Epochs),
-		Balances = utils.deepCopy(Balances),
-		lastKnownCirculatingSupply = LastKnownCirculatingSupply,
-		lastKnownLockedSupply = LastKnownLockedSupply,
-		lastKnownStakedSupply = LastKnownStakedSupply,
-		lastKnownDelegatedSupply = LastKnownDelegatedSupply,
-		lastKnownWithdrawSupply = LastKnownWithdrawSupply,
-		lastKnownRequestSupply = LastKnownPnpRequestSupply,
-		lastKnownTotalSupply = lastKnownTotalTokenSupply(),
-	}
-	local status, resultOrError = pcall(prune.pruneState, msgTimestamp, msgId, LastGracePeriodEntryEndTimestamp)
-	if not status then
-		ao.send({
-			Target = msg.From,
-			Action = "Invalid-Tick-Notice",
-			Error = "Invalid-Tick",
-			Data = json.encode(resultOrError),
-		})
-		Vaults = previousState.Vaults
-		GatewayRegistry = previousState.GatewayRegistry
-		NameRegistry = previousState.NameRegistry
-		Epochs = previousState.Epochs
-		Balances = previousState.Balances
-		msg.ioEvent:addField("Tick-Error", tostring(resultOrError))
-		return true -- stop processing here and return
-	end
+	local resultOrError = prune.pruneState(msgTimestamp, msgId, LastGracePeriodEntryEndTimestamp)
```

Comment:
Why is this `resultOrError` if we're not doing a pcall here?

Reply:
it's more like `prunedStateResult` - updating